pkg/incentives/events.go

```diff
@@ -0,0 +1,119 @@
+// Copyright 2022 The Swarm Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package incentives
+
+import (
+	"context"
+	"sync"
+)
+
+type phaseType int
+
+const (
+	commit phaseType = iota
```

Comment:
Maybe the first one can be `none` so you don't have to initialize it to `-1` in the constructor.